### ISSUE TITLE
`manual_is_multiple_of` should not trigger on literal divisors

### ISSUE THREAD

**Issue by ensc** [rust-lang/rust-clippy#16537]
### What it does

The `manual_is_multiple_of` lint triggers on expressions like

```rust
if v % 2 == 0 { }
```

suggesting replacement with

```rust
if v.is_multiple_of(2) { }
```

Proposal: Skip this lint when the divisor is an integer literal.

### Advantage

- `v % 2 == 0` is an universal idiom across languages and understood immediately
- `v.is_multiple_of(2)` is longer to type and more complex to read
- Main purpose of is_multiple_of() was probably avoiding panics due to division-by-zero; this does not apply to literal divisors (compile-time validation)

### Example

Trigger on

```rust
let a = 2;
if v % a == 0 { }  /// WARNING --> replace by v.is_multiple_of(a)
```

but not

```rust
if v % 2 == 0 { }  /// NO warning
```
